Week4/Lecture8/Assignment3/lecture8_assignment3/src/main/java/com/example/lecture8_assignment3/config/DataSourceConfig.java

```diff
@@ -0,0 +1,60 @@
+package com.example.lecture8_assignment3.config;
+
+import javax.sql.DataSource;
+
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.boot.context.properties.ConfigurationProperties;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.jdbc.core.JdbcTemplate;
+import org.springframework.jdbc.datasource.DataSourceTransactionManager;
+import org.springframework.jdbc.datasource.DriverManagerDataSource;
+import org.springframework.transaction.PlatformTransactionManager;
+import org.springframework.transaction.annotation.EnableTransactionManagement;
+
+@Configuration
+@EnableTransactionManagement
+public class DataSourceConfig {
+
+    @Bean(name = "datasource1")
+    @ConfigurationProperties(prefix = "spring.datasource1")
+    public DataSource dataSource1() {
+        DriverManagerDataSource dataSource = new DriverManagerDataSource();
+        dataSource.setDriverClassName("com.mysql.cj.jdbc.Driver");
```

Comment:
All database info properties should be in properties file and use @Value to get it in java code.

Reply:
I have implemented it, please kindly check, thanks for the feedback!